test/common/http/http3_status_tracker_test.cc

```diff
@@ -0,0 +1,134 @@
+#include "common/http/http3_status_tracker.h"
+
+#include "test/mocks/common.h"
+#include "test/mocks/event/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using Envoy::Event::MockTimer;
+using testing::_;
+using testing::AnyNumber;
+using testing::Return;
+using testing::StrictMock;
+
+namespace Envoy {
+namespace Http {
+
+namespace {
+class Http3StatusTrackerTest : public testing::Test {
+public:
+  Http3StatusTrackerTest() : timer_(&dispatcher_), tracker_(dispatcher_) {}
+
+  testing::NiceMock<Event::MockDispatcher> dispatcher_;
+  testing::StrictMock<MockTimer> timer_;
```

Comment:
CI is failing for tests in this file.  I think that if you run this test under ASAN you'll find that timer_ is being deleted twice.  I think the fix is something like:

```
  Http3StatusTrackerTest() : timer_(new testing::StrictMock<MockTimer>(&dispatcher_)), tracker_(dispatcher_) {}

  testing::NiceMock<Event::MockDispatcher> dispatcher_;
  testing::StrictMock<MockTimer>* timer_;
```